pkg/providerConfigs/detokenize.go

```diff
@@ -7,308 +7,197 @@ See the LICENSE file for more details.
-	err := filepath.Walk(path, detokenizeAdditionalPath(path, tokens))
-	if err != nil {
-		return err
-	}
+func setMetaphorTokens(tokens *MetaphorTokenValues, gitProtocol string) {
+	tokens.GitProtocol = gitProtocol
+	tokens.MetaphorDevelopmentIngressURL = fmt.Sprintf("https://metaphor-dev.%s", tokens.DomainName)
+	tokens.MetaphorProductionIngressURL = fmt.Sprintf("https://metaphor.%s", tokens.DomainName)
+	tokens.MetaphorStagingIngressURL = fmt.Sprintf("https://metaphor-stage.%s", tokens.DomainName)
 
-	return nil
 }
 
-// detokenizeAdditionalPath temporary addition to handle detokenizing additional files
-func detokenizeAdditionalPath(path string, tokens *GitopsDirectoryValues) filepath.WalkFunc {
-	return filepath.WalkFunc(func(path string, fi os.FileInfo, err error) error {
-		if err != nil {
-			return err
+func setGitOpsTokens(tokens *GitopsDirectoryValues, gitProtocol string, useCloudflareOriginIssuer bool) {
+	if tokens.SubdomainName != "" {
+		if !strings.Contains(tokens.DomainName, tokens.SubdomainName) {
+			tokens.DomainName = fmt.Sprintf("%s.%s", tokens.SubdomainName, tokens.DomainName)
 		}
+	}
+	//origin issuer defines which annotations should be on ingresses
+	if useCloudflareOriginIssuer {
+		tokens.CertManagerIssuerAnnotation1 = "cert-manager.io/issuer: cloudflare-origin-issuer"
+		tokens.CertManagerIssuerAnnotation2 = "cert-manager.io/issuer-kind: OriginIssuer"
+		tokens.CertManagerIssuerAnnotation3 = "cert-manager.io/issuer-group: cert-manager.k8s.cloudflare.com"
+		tokens.CertManagerIssuerAnnotation4 = "external-dns.alpha.kubernetes.io/cloudflare-proxied: \"true\""
+	} else {
+		tokens.CertManagerIssuerAnnotation1 = "cert-manager.io/cluster-issuer: \"letsencrypt-prod\""
+	}
 
-		if !!fi.IsDir() {
-			return nil
-		}
-
-		// var matched bool
-		matched, _ := filepath.Match("*", fi.Name())
-
-		if matched {
-			// ignore .git files
-			if !strings.Contains(path, "/.git/") {
-				read, err := ioutil.ReadFile(path)
-				if err != nil {
-					return err
-				}
+	if tokens.CloudProvider == "k3s" {
+		tokens.K3sEndpoint = tokens.K3sServersPrivateIps[0]
+	}
 
-				newContents := string(read)
-				newContents = strings.Replace(newContents, "<GITLAB_OWNER>", tokens.GitlabOwner, -1)
+	// The fqdn is used by metaphor/argo to choose the appropriate url for cicd operations.
+	if gitProtocol == "https" {
+		tokens.GitFqdn = fmt.Sprintf("https://%v.com/", tokens.GitProvider)
+	} else {
+		tokens.GitFqdn = fmt.Sprintf("git@%v.com:", tokens.GitProvider)
+	}
+}
 
-				err = ioutil.WriteFile(path, []byte(newContents), 0)
-				if err != nil {
-					return err
-				}
-			}
-		}
-		return nil
-	})
+// renderGoTemplating - Renders the template with the given values
+// it also includes the sprig GenericFuncMap functions listed here: https://masterminds.github.io/sprig/.
+func parseTemplate(content string) (*template.Template, error) {
+	funcs := sprig.GenericFuncMap()
+	funcs["toJson"] = toJson
```

Comment:
Masterminds/sprig includes a `toJson` function already and the function below seems generic enough